docs/sources/operators-guide/configure/configuring-hash-rings.md

```diff
@@ -77,6 +77,25 @@ If you run multiple Mimir processes on the same node or the port `7946` is not a
 - `-memberlist.advertise-addr`: IP address to advertise to other Mimir replicas. The other replicas will connect to this IP to talk to the instance.
 - `-memberlist.advertise-port`: Port to advertise to other Mimir replicas. The other replicas will connect to this port to talk to the instance.
 
+#### Cluster label verification
+
+By default, Grafana Mimir memberlist will join a cluster with any instance discovered when resolving the hosts configured via the `-memberlist.join` CLI flag (or its respective YAML configuration option).
+If, for any reason, the discovered addresses include instances of other Grafana Mimir clusters, or instances of other distributed systems using memberlist, Grafana Mimir will join these unrelated clusters together.
+
+To avoid this type of undesired situation Grafana Mimir offers an additional type of validation known as cluster label verification.
+
+When enabled, all memberlist internal traffic will be prefixed with the configured cluster label, discarding any other traffic that doesn't match that prefix and ensuring that only replicas having the same configured label can connect to each other.
+
+#### Migrate to using cluster label verification
+
+To migrate a Grafana Mimir cluster to use cluster label verification, the following steps need to be performed:
+
+- Disable cluster label verification on all cluster instances via `-memberlist.cluster-label-verification-disabled` CLI flag (or its respective YAML configuration option).
```

Comment:
A couple of notes:
1. Make it a numbered list
2. Clarify that they need to rollout these changes 1-by-1, not all of them together. For example you could intersticial steps like this:

```
1. Disable cluster label verification on all cluster instances ......
1. Wait until the configuration change has been rolled out to all Mimir instances.
1. Define a label on all cluster instances which is unique to the cluster ......
1. Wait until the configuration change has been rolled out to all Mimir instances.
```